src/redis-cli.c

```diff
@@ -9209,13 +9338,31 @@ static void findBigKeys(int memkeys, unsigned memkeys_samples) {
         freeReplyObject(reply);
     } while(force_cancel_loop == 0 && it != 0);
 
+    /* Final progress bar if TTY */
+    if (IS_TTY_OR_FAKETTY()) {
+        displayKeyStatsProgressbar(sampled, total_keys);
+
+        /* Clean the types info shown during the progress bar */
+        int line_count = 0;
+        di = dictGetIterator(types_dict);
+        while ((de = dictNext(di)))
+            line_count += cleanPrintfln("");
+        dictReleaseIterator(di);
+        printf("\033[%dA\r", line_count);
+    }
+
     if(types) zfree(types);
     if(sizes) zfree(sizes);
 
     /* We're done */
     printf("\n-------- summary -------\n\n");
-    if (force_cancel_loop) printf("[%05.2f%%] ", pct);
-    printf("Sampled %llu keys in the keyspace!\n", sampled);
+
+    /* Show percentage and sampled output when writing to a file */
+    if (!IS_TTY_OR_FAKETTY()) {
+        if (force_cancel_loop) printf("[%05.2f%%] ", pct);
+        printf("Sampled %llu keys in the keyspace!\n", sampled);
+    }
```

Comment:
why not showing this in the TTY?

Reply:
In the TTY, It is already displayed as part of the progress bar:
```
 16.62% ||||||||||--------------------------------------------------
Keys sampled: 8315
```
We preserved the old output when redirected to a file in case someone did an automation around that.